compositor/xwayland.c

```diff
@@ -138,13 +139,28 @@ static void wc_xwayland_new_surface(struct wl_listener *listener, void *data) {
 	wl_list_insert(&server->views, &view->link);
 }
 
+static void wc_xwayland_ready(struct wl_listener *listener, void *data) {
+	struct wc_server *server =
+			wl_container_of(listener, server, xwayland_ready);
+	if (server->startup_cmd != NULL) {
+		// NOTE Executed here so that DISPLAY is correct for the client
+		wlr_log(WLR_INFO, "Executing \"%s\"", server->startup_cmd);
+		if (fork() == 0) {
+			execl("/bin/sh", "/bin/sh", "-c", server->startup_cmd, NULL);
```

Comment:
Some BSD people will eventually argue that `env` is better. I never saw a system where the POSIX shell isn't in /bin.

Reply:
`execlp` would be a simpler way to do that (involves starting one process less).